inc/math/linalg.hpp

```diff
@@ -379,20 +381,100 @@ namespace exo
 				return r;
 			}
 
-			template <ssize_t MC>
-			Vec<S, R> operator* (Vec<S, MC>& v)
+
+			Vec<S, R> operator* (Vec<S, C>& v) { return *this * std::move(v); }
+			Vec<S, R> operator* (Vec<S, C>&& v)
 			{
 				Vec<S, R> r;
 
 				for (int row = R; row--;)
-				for (int col = MC; col--;)
+				for (int col = C; col--;)
 				{
 					r[row] += this->m[row][col] * v[col];
 				}
 
 				return r;
 			}
 
+			Mat<S, R, C> operator* (S s)
```

Comment:
Why no const ref on these operator args `(S s )`?

Reply:
Because S is a storage type, in most cases it will be a simple numerical type. Passing a reference would most likely be less efficient, and it would prevent us from being able to use a literal, like `M * 2.0`

Reply:
Ah. Yep I agree with you. I was in matrix mode.

Reply:
Agreed. I was still in matrix mode.